src/abilities/Golden-Wyrm.js

```diff
@@ -233,78 +233,133 @@ export default (G) => {
-				crush: 5,
-			},
-			_targetTeam: Team.enemy,
+			_targetTeam: Team.ally,
+
+			_maxTransferAmount: 50,
 
-			// 	require() :
 			require: function () {
 				if (!this.testRequirements()) {
 					return false;
 				}
 
-				let map = G.grid.getHexMap(
-					this.creature.x - 2,
-					this.creature.y - 2,
-					0,
-					false,
-					matrices.frontnback2hex,
-				);
-				// At least one target
+				/* Golden Wyrm needs to be left with at least 1 hp after the ability. We
+				can't use ability.costs because that is a fixed value, whereas this ability
+				has a dynamic health cost. */
+				if (this.creature.health < this._maxTransferAmount + 1) {
+					this.message = 'Not enough health';
+					return false;
+				}
+
 				if (
-					!this.atLeastOneTarget(map, {
+					!this.atLeastOneTarget(this._getHexes(), {
 						team: this._targetTeam,
+						optTest: this._confirmTarget,
 					})
 				) {
 					return false;
 				}
+
 				return true;
 			},
 
-			// 	query() :
 			query: function () {
-				let ability = this;
-				let wyrm = this.creature;
+				const ability = this;
+				const wyrm = this.creature;
 
 				G.grid.queryCreature({
 					fnOnConfirm: function () {
 						ability.animation(...arguments);
 					},
+					optTest: this._confirmTarget,
 					team: this._targetTeam,
 					id: wyrm.id,
 					flipped: wyrm.flipped,
-					hexes: G.grid.getHexMap(wyrm.x - 2, wyrm.y - 2, 0, false, matrices.frontnback2hex),
+					hexes: this._getHexes(),
 				});
 			},
 
-			//	activate() :
 			activate: function (target) {
-				let ability = this;
-				ability.end();
+				this.end();
 
-				let damage = new Damage(
-					ability.creature, // Attacker
-					ability.damages, // Damage Type
-					1, // Area
-					[], // Effects
-					G,
+				/* The amount of health transferred is the creature's missing life, capped 
+				to 50. */
+				const transferAmount = Math.min(
+					target.stats.health - target.health,
+					this._maxTransferAmount,
 				);
-				target.takeDamage(damage);
 
-				//remove frogger bonus if its found
-				ability.creature.effects.forEach(function (item) {
-					if (item.name == 'Offense++') {
-						item.deleteEffect();
-					}
-				});
+				target.heal(transferAmount);
+
+				/* Damage Golden Wyrm using `.heal()` instead of `.takeDamage()` to apply 
+				raw damage that bypasses dodge, shielded, etc and does not trigger further 
+				effects. */
+				this.creature.heal(-transferAmount);
+
+				// TODO: Should we show messages/hints for this transfer beyond health lost/gained?
+
+				if (this.isUpgraded()) {
+					const regrowthBuffEffect = new Effect(
+						this.title,
+						this.creature,
+						this.creature,
+						'',
+						{
+							alterations: {
+								regrowth: 10,
+							},
+						},
+						G,
+					);
+
+					this.creature.addEffect(
+						regrowthBuffEffect,
+						// TODO: What should this say?
+						`%CreatureName${this.creature.id}% is brimming with health`,
+						// TODO: Should we show a hint?
```

Comment:
@DreadKnight There's currently no hint when the upgraded ability triggers. Should there be one?

Reply:
Well, there's a visual "upgrade" animation on the ability icons themselves plus they gain a cooler frame around.
But I guess we could have a message log indicating that an ability has upgraded as well, would make way more sense in some cases while looking around through the log.

`Visible Stigmata has been upgraded`

Reply:
@DreadKnight Sounds good, but can you please creature a separate issue as that change should be applied across the board to other creatures.

Reply:
@nandastone Good point. Here it is -> #1962